e2e/cypress/integration/messaging/emoji_to_markdown_spec.js

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+function createMessages(message) {
+    const aliasLineMessage = 'newLineMessage';
+    const aliasLineMessageReplyInRHS = 'newLineMessageReplyInRHS';
+
+    cy.postMessage(message);
+    cy.getLastPostId().then((postId) => {
+        cy.get(`#postMessageText_${postId}`).as(aliasLineMessage);
+        cy.clickPostCommentIcon(postId);
+    });
+
+    cy.postMessageReplyInRHS(message + '{enter}');
+    cy.getLastPostId().then((postId) => {
+        cy.get(`#postMessageText_${postId}`).as(aliasLineMessageReplyInRHS);
+    });
+
+    return [aliasLineMessage, aliasLineMessageReplyInRHS];
+}
+
+function hasMarkdown(message) {
+    const aliases = createMessages(message);
+
+    aliases.forEach((alias) => {
+        cy.get('@' + alias).
+            children().should('have.class', 'post-code').
+            children('code').should('be.visible').contains(message.trim());
+    });
+}
+
+function hasNoMarkdown(message) {
+    const aliases = createMessages(message);
+
+    aliases.forEach((alias) => {
+        cy.get('@' + alias).
+            children().should('not.have.class', 'post-code').
+            children('code').should('have.length', 0);
```

Comment:
Instead of checking as not code, I think this should be verified if rendered as (jumbo) emoji, something like:
```javascript
cy.get('@' + alias).
    children().should('have.class', 'all-emoji').
    children().find('span').last().should('have.class', 'emoticon').and('have.attr', 'title', message.trim() === ':D' ? ':smile:' : ':taco:');
```

Reply:
outdated, please have another look.